src/microsoft.sheets/spreadsheet_endpoint.bal

```diff
@@ -0,0 +1,579 @@
+        }
+
+        //If the worksheet was created we will get a JSON response with the newly created worksheet's information
+        json|error responseJson = httpResponse.getJsonPayload();
+
+        if !(responseJson is map<json>) {
+            typedesc<any|error> typeOfRespone = typeof responseJson;
+            return TypeConversionError("Invalid response; expected a `map<json>` found " + typeOfRespone.toString());
+        }
+
+        map<json> responsePayload = <map<json>>responseJson;
+
+        json|error identifier = responsePayload.id;
+
+        if !(identifier is string) {
+            typedesc<any|error> typeOfIdentifier = typeof identifier;
+            return TypeConversionError("Invalid identifier; expected a `string` found " + typeOfIdentifier.toString());
+        }
+
+        string sheetId = <string>identifier;
+
+        json|error sheetPosition = responsePayload.position;
+
+        if !(sheetPosition is int) {
+            typedesc<any|error> typeOfPosition = typeof sheetPosition;
+            return TypeConversionError("Invalid sheet position; expected an `int` found " + typeOfPosition.toString());
+        }
+
+        int position = <int>sheetPosition;
+
+        //Populate a new worksheet object using the information from the properties as well as from the received JSON object
+        Worksheet workSheet = new (self.microsoftGraphConfig, self.properties.path,
+            self.properties.workbookName, sheetId, worksheetName, position);
+        return workSheet;
+    }
+
+    # Remove a worksheet from this workbook.
+    # + worksheetName - name of the worksheet to be removed
+    # + return - boolean true on success, else returns an error
+    public remote function removeWorksheet(string worksheetName) returns @tainted Error? {
+        http:Request request = new;
+        http:Response|error httpResponse = self.workbookClient->delete("/v1.0/me/drive/root:/" +
+            self.properties.workbookName + ".xlsx:/workbook/worksheets/" + worksheetName, request);
+
+        if (httpResponse is http:Response) {
+            if (httpResponse.statusCode == http:STATUS_NO_CONTENT) {
+                return ();
+            } else {
+                return HttpResponseHandlingError("Error occurred while deleting the worksheet.");
+            }
+        } else {
+            return HttpError("Error occurred while accessing the Microsoft Graph API.", httpResponse);
+        }
+    }
+};
+
+# Worksheet Client Object.
+# + worksheetClient - HTTP client endpoint for the worksheet
+# + properties - worksheet specific properties
+# + microsoftGraphConfig - Configurations for accessing spreadsheet API
+public type Worksheet client object {
+    http:Client worksheetClient;
+    WorksheetProperties properties;
+    MicrosoftGraphConfiguration microsoftGraphConfig;
+
+    public function init(MicrosoftGraphConfiguration msGraphConfig, string path, string workbookName, string sheetId,
```

Comment:
Same as before, can't we pass the `httpClient` we created before

Reply:
We can. Refactored the code to use the same HTTP client across all the client objects.